src/utils/spo.ts

```diff
@@ -1453,5 +1453,37 @@ export const spo = {
     const webProperties: WebProperties = await request.get<WebProperties>(requestOptions);
 
     return webProperties;
+  },
+
+  /**
+  * removes a file by url
+  * @param webUrl The web url
+  * @param url The url of the file
+  * @param recycle If file should be recycled or not
+  * @param logger The logger object
+  * @param verbose Set for verbose logging
+  */
+  async removeFile(webUrl: string, url: string, recycle?: boolean, logger?: Logger, verbose?: boolean): Promise<void> {
```

Comment:
@nicodecleyre the main issue of this change is that this util method introduced duplicated logic/code that is present also here 
https://github.com/pnp/cli-microsoft365/blob/a7465c9abfe38e7bcba28f1979377a7520037c7b/src/m365/spo/commands/file/file-remove.ts#L125-L142

In order to introduce less logic we need to maintain we should also reuse this util method in the `file-remove.ts` as well before we proceed